### PR TITLE
Improves realtime widget

### LINKED ISSUE #12876: See IP addresses in Real-time visitor widget
After update in section "Visitors in Real-time" IPs of user disappear  I can't see users IPs.
This function is not longer support or I need enable it somewhere?

### LINKED ISSUE #12887: Display visitor provider in Real time widget
It looks like the new visitor log widget doesn't show geoip provider information, when hovering on the flag as it used to. Can we have it back please? :)

### PR DESCRIPTION
**Pull request by sgiehl**
* permanently shows link to visitor profile in new line, with user id if available, otherwise visitor id is shown
* adds visitor IP to country flag tooltip
* readds provider name to country flag tooltip (if available)
* improves styling 

*Note: the tooltip changes will also have effect in visitor log & profile*

Changes will look like this:

image

fixes #12876 
fixes #12887

### CODE REVIEW

**Comment by mattab** [comment]
Feedback:

* when User ID is defined, it's great to display it, and when it's not defined we don't want to show visitor id. So when User ID is defined, maybe it's acceptable to introduce a new line and write the User ID on the new line, but maybe first try display User ID on the right of the icon of Visitor profile? 
* let's not use a new line for the visitor profile. It should be fine to update the icon on the first line to the font icon... or alternatively add it to the second line with all the other icons
* Also another unrelated feedback: the space between the line of icons and the search engine row, is taller than the space between datetime and icons. Let's make all spacing consistent. Also let's check at different resolutions, and different column layouts, as it can change the look of the widget.

**Comment by sgiehl** [comment]
> when User ID is defined, it's great to display it, and when it's not defined we don't want to show visitor id. So when User ID is defined, maybe it's acceptable to introduce a new line and write the User ID on the new line, but maybe first try display User ID on the right of the icon of Visitor profile?

I decided to add the visitorid in order to have a bit more consistency. We already displayed the visitor profile icon and the user id before. But to be honest, I never recognized that it has been displayed. The small icon behind the time was quite invisible. Imho it's ugly from UI/UX perspective if the icon/link can be found on different places (behind the date or on new line depeding on the available space). Now it's always at the same position and easy to find and click.

> let's not use a new line for the visitor profile. It should be fine to update the icon on the first line to the font icon... or alternatively add it to the second line with all the other icons

The icons displayed here are generated with the same code as in visitor log and profile. I really would like to keep the consistency between all widgets/reports, so it's clear to the user which information can be found within the icon list. Imho that improves the UX a lot. Even if we already added various special handling for log/profile/widget so specific icons are not displayed in some parts, I would avoid changing more stuff here.

**Comment by diosmosis** [comment]
👍 to show visitor id if no userid, would confuse me otherwise to see it one place and not the other. Though maybe it would be clearer to have '(Unknown user)' or something instead? Just some thoughts.

**Comment by mattab** [comment]
>  Imho it's ugly from UI/UX perspective if the icon/link can be found on different places

Agreed with this. That said, we can't/don't want to display the visitor ID as this is meaningless and not human readable. There is no value in showing it in the widget. 

So maybe just show the User ID after the list of icons, when it's defined? And show the Visitor Profile icon in the list of icons?

**Comment by sgiehl** [comment]
Alternatively we could show the text `open visitorprofile` or similar instead of the visitor id.
I'm not sure if moving the icon to the other icons is a good option. To have a consistency in UX/UI we should imho always try to show the same icons in visitor log / profile & real time widget.

**Comment by mattab** [comment]
Can you try moving the icon either at the end of other icons, or on the date-time row? I think it will be fine... 

(it's not critical to have consistency between the widgets but rather we want to give the best possible design / data presentation for this widget)

I wouldn't show the text "open visitor profile" because it's too wordy for the real time widget which is space constrained.

**Comment by tsteur** [comment]
It may still be useful to show the visitorId on hover for debugging purposes. Eg someone is visiting your website, and then you want to segment for that visitor, or use the GDPR tools to segment for that visitor etc. But not super important needed for sure.

**Comment by mattab** [comment]
On hover sounds good :+1:

**Comment by sgiehl** [comment]
Ok. So what exactly shall we do now? If we move the icon for the visitor profile somewhere else, the user id will not be displayed next to it anymore as there is not enough space to have it always in the same line. So the userid will be displayed in an extra line (if defined), but not linked with the visitor profile.

And if so, where shall we display the visitorid on hover? The icon already has a tooltip saying "open visitor profile". And using a tooltip for the visitorid imho isn't a good solution as tooltip contents can't be copied.

**Comment by mattab** [comment]
I'm not sure how it will look, but thinking to display the Visitor profile icon after the other icons (or on the date-time line...). 

For User ID, maybe display it after date-time, or on a new line would be fine (since it is an important info and will only be set quite rarely). Ideally the User ID also links to open the visitor profile (if possible.)

Visitor ID on hover: in the same tooltip as the Visitor IP address

**Comment by sgiehl** [comment]
@mattab I've done the changes

**Comment by diosmosis** [comment]
@sgiehl the positioning of the icon looks odd when I tried it locally:

image

Tried quickly fixing it, but couldn't figure out an easy way to do it (changing the size is easy, but couldn't find a way to move it down).

**Comment by sgiehl** [comment]
sure. should be fixed now

**Comment by mattab** [comment]
Looks really good :+1:
